pubsub/pubsub-aws/src/main/java/com/salesforce/multicloudj/pubsub/aws/AwsSubscription.java

```diff
@@ -443,6 +454,15 @@ public Builder withSqsClient(SqsClient sqsClient) {
             return this;
         }
         
+        /**
+         * Directly set the subscription URL to avoid calling GetQueueUrl again.
+         * Used when the queue URL has already been resolved 
+         */
+        Builder withSubscriptionUrl(String subscriptionUrl) {
```

Comment:
who is the target user of this function? I am asking because end user only should have one consistent way across all clouds. If this is just for tests, we shouldn't expose it here.